### PR TITLE
feat-auto-ssl-generation

### PR DESCRIPTION
**Pull request by lohanidamodar**
## What does this PR do?

(Provide a description of what this PR does.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: app/http.php [hunk 1]
```diff
@@ -65,22 +66,36 @@
...
 
-$domain = App::getEnv('_APP_DOMAIN', '');
-
-Console::info('Issuing a TLS certificate for the master domain ('.$domain.') in 30 seconds.
-    Make sure your domain points to your server IP or restart your Appwrite server to try again.'); // TODO move this to installation script
+function certificateSetup($domain)
+{
+    if($domain=='localhost' || (bool)ip2long($domain)) return;
+    $domains = Config::getParam('domains', []);
+    if (!in_array($domain, $domains)) {
```
Comment (eldadfux) [inline app/http.php:73]: array_key_exists check will be faster than in_array. You can have an array as follow:
[
    'example.com' => true,
    'example2.com' => true,
    'example3.com' => true,
]

File: app/http.php [hunk 2]
```diff
@@ -65,22 +66,36 @@
...
-    'validateCNAME' => false,
-]);
+}
 
 $http->on('request', function (SwooleRequest $swooleRequest, SwooleResponse $swooleResponse) {
     $request = new Request($swooleRequest);
     $response = new Response($swooleResponse);
 
+    $domain = $request->getHostname();
+    certificateSetup($domain);
```
Comment (eldadfux) [inline app/http.php:97]: If we're using this function only once, we can skip it and place the code here.

**Review by eldadfux — CHANGES_REQUESTED** [review]

File: app/controllers/general.php [hunk 3]
```diff
@@ -65,22 +63,21 @@
...
                     'validateTarget' => false,
                     'validateCNAME' => false,
                 ]);
             }
 
-            $checkedDomains[$domain->get()] = true;
+            $domains[$domain->get()] = true;
 
         }
         Console::info('adding ' . $domain->get() . ' to list of domains already checked');
```
Comment (eldadfux) [inline app/controllers/general.php:79]: We can probably skip this log as well

File: app/controllers/general.php [hunk 4]
```diff
@@ -14,24 +14,71 @@
...
+                    ],
+                    'domain' => $domain->get(),
+                ];
+                $dbDomain = $consoleDB->createDocument($dbDomain);
+                Authorization::enable();
+
+                Console::info('Issuing a TLS certificate for the master domain (' . $domain->get() . ') in ~30 seconds..'); // TODO move this to installation script
+
+                ResqueScheduler::enqueueAt(\time() + 30, 'v1-certificates', 'CertificatesV1', [
+                    'document' => $dbDomain,
```
Comment (eldadfux) [inline app/controllers/general.php:69]: can you verify this is actually working, not sure if we must use ->getArrayCopy() here

Comment (lohanidamodar) [reply app/controllers/general.php:69]: Its working

**Review by eldadfux — APPROVED** [review]

**Review by kodumbeats — APPROVED** [review]
👍🏻

**Pull request merged**